sources/packages/backend/apps/api/src/route-controllers/application/application.aest.controller.ts

```diff
@@ -64,19 +64,34 @@ export class ApplicationAESTController extends BaseController {
   /**
    * API to fetch application details by applicationId.
    * This API will be used by ministry users.
-   * @param applicationId
+   * @param applicationId application id.
+   * @param loadDynamicData flag for if dynamic data should be loaded.
+   * @param isParentApplication flag for if the application is a parent application.
    * @returns Application details
    */
   @Get(":applicationId")
-  @ApiNotFoundResponse({ description: "Application not found." })
+  @ApiNotFoundResponse({
+    description:
+      "Application not found or current application for provided parent application not found.",
+  })
   async getApplication(
     @Param("applicationId", ParseIntPipe) applicationId: number,
     @Query("loadDynamicData", new DefaultValuePipe(true), ParseBoolPipe)
     loadDynamicData: boolean,
+    @Query("isParentApplication", new DefaultValuePipe(false), ParseBoolPipe)
+    isParentApplication: boolean,
   ): Promise<ApplicationSupplementalDataAPIOutDTO> {
+    const currentApplicationId =
```

Comment:
Please add a comment here similar to this to explain what is happening. 
```
// When the application is a parent application, get the current application by parent application id.
 // Otherwise, set the current application id to the provided application id.
```